cluster-autoscaler/core/scale_down.go

```diff
@@ -401,17 +401,50 @@ func (sd *ScaleDown) CleanUpUnneededNodes() {
 // * timestamp is the current timestamp.
 // * pdbs is a list of pod disruption budgets.
 func (sd *ScaleDown) UpdateUnneededNodes(
-	allNodes []*apiv1.Node,
```

Comment:
Why does TryToScaleDown take allNodes and pods as params, why UpdateUnneededNodes takes it from snapshot? This seems really inconsistent for no good reason.

Reply:
Yeah. Seems like I need spend more than a few minutes to cleanup mess in scale-down. Would that be fine to do this as a followup?

Reply:
I added it as a followup at very top of the list

Reply:
I did not realize it was an ordered list :p

Reply:
Also - you did not. Please do, I really don't want it to stay that way.

Reply:
Damn - I must have not click save or something. I am sure I was writing the bullet point.